### PR TITLE
net: lib: coap: update method_from_code() to return error.

### LINKED ISSUE #49498: net: lib: coap: update method_from_code() to report success/failure
**Is your enhancement proposal related to a problem? Please describe.**
`coap_handle_request()` is unable to distinguish between two different situations:
(1) the request code is invalid (other than GET, PUT, POST, DEL, etc.)
(2) the function pointer is NULL for a valid request code

This is because `method_from_code()` returns NULL regardless of the two cases mentioned above.

Since `coap_handle_request()` is used in the COAP server which should respond to the client with different response codes in these situations, `method_from_code()` should be updated.
 
**Describe the solution you'd like**
The API for `method_from_code()` should be changed to report status (success/failure) separately from the function pointer.
e.g. 
`int method_from_code(const struct coap_resource *resource, uint8_t code, coap_method_t *method)`

**Describe alternatives you've considered**
None.

**Additional context**
Add any other context or graphics (drag-and-drop an image) about the feature request here.

### PR DESCRIPTION
**Pull request by sshahleg**
method_from_code() function signature changed to return status and method.
In case of invalid request code, it returns -EINVAL which causes coap_handle_request()
to return -ENOTSUP. Adding tests as well.

Fixes: #49498

Signed-off-by: Sagar Shah <sagar.shah@legrand.com>

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]
Thanks!

**Comment by sshahleg** [comment]
@carlescufi , how do I bring back the other reviewers? I was re-requesting a review from you, but didn't realize it would drop the others.

File: subsys/net/lib/coap/coap.c [hunk 1]
```diff
@@ -819,26 +819,33 @@ static bool uri_path_eq(const struct coap_packet *cpkt,
 	return true;
 }
 
-static coap_method_t method_from_code(const struct coap_resource *resource,
-				      uint8_t code)
+static int method_from_code(const struct coap_resource *resource,
+				      uint8_t code, coap_method_t *method)
```
Comment (pdgendt) [inline subsys/net/lib/coap/coap.c:823]: Nit: can you fix the alignment

**Review by pdgendt — DISMISSED** [review]

**Review by rlubos — COMMENTED** [review]
Looks good but please fix the alignment as suggested.

**Review by carlescufi — DISMISSED** [review]

**Comment by carlescufi** [comment]
> @carlescufi , how do I bring back the other reviewers? I was re-requesting a review from you, but didn't realize it would drop the others.

No worries, fix the alignment issues and then we'll re-request reviews.

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**